src/components/Modals/CerificateModal.tsx

```diff
@@ -24,144 +24,332 @@ interface Props {
   reloadCertificates: () => void;
 }
 
-const STEPS = 4;
-
 export interface Activity {
   index: number;
   text: string;
 }
 
 export interface CertificateData {
-  student?: string;
+  pupil?: string;
   endDate: number;
-  weekCount: number;
   hoursPerWeek: number;
   hoursTotal: number;
   subjects: string[];
-  mediaType: string | null;
+  medium: string | null;
   activities: string[];
-  ongoingLessons: boolean;
-  lang: ISupportedLanguage;
+  ongoingLessons?: boolean;
+  automatic?: boolean;
 }
 
 const CertificateModal: React.FC<Props> = ({ user, reloadCertificates }) => {
-  const [loading, setLoading] = useState(false);
-  const allMatches = [...user.matches, ...user.dissolvedMatches];
+  const modalContext = useContext(context.Modal);
+  const [createdCertificate, createCertificate, reset] = useAPICallable(
+    'createCertificate'
+  );
+
+  // If the certificate was created, the cards need to be reloaded on the Settings page:
+  useEffect(() => {
+    if (createdCertificate.value) reloadCertificates();
+  }, [createdCertificate]);
+
+  return (
```

Comment:
An abort button or something similar would be nice here. If for example an error occurs I cannot simply leave the modal 😉